consent/handler.go

```diff
@@ -588,6 +588,10 @@ func (h *Handler) getOAuth2ConsentRequest(w http.ResponseWriter, r *http.Request
 		request.RequestedAudience = []string{}
 	}
 
+	if request.Client.SkipConsent {
+		request.Skip = true
```

Comment:
This typically means that there is an existing, previous consent for that user & app. In this case though it says: just skip this!

I'm not 100% sure but it's possible that `Skip==true` has other implications (e.g. remember can not also be true?).

Reply:
i've been thinking about this too - should we actually record an auto-granted consent rather than skipping it entirely (assuming i'm reading this correctly)?
otoh, it's a per-client setting, so remember/not remember has no real effect as a new consent would also just be granted. Only if the users turn off skipConsent later, this matters, and in that case i'd argue it's actually better to not consider the auto-granted consents in the future.

Reply:
What the consent app does depends on it's implementation, but both hydra-login-consent-node (https://github.com/ory/hydra-login-consent-node/blob/master/src/routes/consent.ts#L39) as well as the consent we use in the Ory Network return early by accepting the consent with all requested scopes on behalf of the user.

I agree that the semantics are different between "skip this for this user because consent was already given" and "skip this for all users because consent is not necessary", but the result and the code path is identical.

Reply:
Regarding analytics, we can surely emit events when a consent was skipped.

Reply:
Ok I see. In that case, it probably makes sense to maybe just add this logic to the consent app where it checks for the client field? Then, I think the semantics would be cleaner and the code is minimal :)

Reply:
We can do that! I added the check in https://github.com/ory/hydra-login-consent-node/pull/137 (and will add in Cloud).

I still want to check that the consent is skipped in Hydra E2E, so we need to merge the logic in the consent Node app first.

Reply:
Done, and tested with hydra-consent-node